Removed/Fixed W9 upload  + Bug Fixes + Dark Mode

jsnowfreedman opened the pull request:
Removed/Fixed W9 Files uploads. Members should no longer be required to upload a w9 directly, but use a different process

Fix some development/runtime issues/pinned library.

Working on getting dark mode enabled for the calendar.

**Dark mode could probably be done better... but it was burning my retinas at 2am, so I decided to fix it at 2am... + The other changes where because I was pestered. I make no guarantees over the quality of this code :)**

jsnowfreedman committed:
Bug Fixes / Best practice patches

Added proper shebang to shell scripts.
Fixed XDebug version pin.
EmailComponent.php argument issue fix.

jsnowfreedman committed:
Started a dark mode

Only works on the main calendar page atm, should be easy to get onto the other pages though.
Added some JS to make it save the last dark/light mode, or prefer OS if nothing else. All else fails, it will prefer light mode.

excalbian reviewed (approved):
Looks good from a read-through (just drinking my first cup o joe though). Not able to run this right now, so I'm assuming the modernization changes (`/usr/bin/env`) are supported in the docker image.

jsnowfreedman commented:
@excalbian yep. I actually had to change them in order to get them working on the images.

The xdebug version pinning was also a requirement to run/test - a new version of xdebug has been released to pecl that was no longer compatible with our version of PHP.

jsnowfreedman commented:
What dark/night mode looks like:
![image](https://user-images.githubusercontent.com/59654518/215828557-e0eb92ca-ca9c-42b2-8907-85b985697a86.png)

mandarl commented on webroot/css/app.css:
```diff
@@ -265,4 +265,96 @@ footer {
+    visibility: hidden;
+}
+
+#day-night-slider > label {
```
It will be better to use custom css properties for the colors so you won't have to duplicate entire classes.
It would look something like this:
```
:root {
  --bg-color: #fff;
  --h1-color: #333;
  --text-color: #000;
}

[data-theme="dark"] {
  --bg-color: #131516;
  --h1-color: #c8c3bc;
  --text-color: #9f978b;
}

h1, h2, h3, h4, h5, legend {
  color: var(--h1-color) !important;
}

label, div, textarea, select, input {
  color: var(--text-color) !important;
}

```

mandarl commented on webroot/js/app.js:
```diff
@@ -217,6 +217,66 @@ $(function() {
+          if(this.checked) {
+            $('body').addClass("dark");
+            $('#day-night-slider > label > span')[0].innerHTML = "Night";
+            setCookie("lastChosenColorScheme", "dark", 30);
```
nit: use localStorage instead of cookie since cookies are sent to the server with each request while this property is only used on the client.

jsnowfreedman replied:
So the cookies versus local storage thing was actually because I was debating making some of the other apps also integrate with the same dark selection...

Realistically, I ain't going to get to those for a while though, so it's probably fine to just use local storage for now. (Also, there's the fact that there's no logic anywhere on any of the back ends to actually save this or anything along those lines so -shrug-)

mandarl replied:
Ah .. that makes sense! .. maybe in the future we can have a separate service for user preferences across all DMS apps 😊

jsnowfreedman replied:
Yeah, as I was kind of rereading this code just now and thinking about it- that'll probably make the most sense. There's some work that I've put into making a service that could handle that for DMS, so I guess whenever that happens I can revisit it.

I'll go ahead and get this using local storage instead for now later tonight

mandarl reviewed (commented):
Looks good other than nits 👍

jsnowfreedman committed:
W9 Fixes, Dark Mode, Image Preview, Cleanup

Removed W9 upload options and replaced help text for user to email to proper place.
Changed dark mode settings to localStorage and moved slider to top bar.
Updated to curly braced PHP syntax for at least one file... More to follow.
Added image preview tabs for viewing events.

jsnowfreedman commented:
Split into: #150, #151, #152, #153
